flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/util/UpdatingPlanChecker.scala

```diff
@@ -39,14 +40,21 @@ object UpdatingPlanChecker {
   }
 
   /** Extracts the unique keys of the table produced by the plan. */
-  def getUniqueKeyFields(plan: RelNode): Option[Array[String]] = {
-    getUniqueKeyGroups(plan).map(_.map(_._1).toArray)
-  }
-
-  /** Extracts the unique keys and groups of the table produced by the plan. */
-  def getUniqueKeyGroups(plan: RelNode): Option[Seq[(String, String)]] = {
-    val keyExtractor = new UniqueKeyExtractor
-    keyExtractor.visit(plan)
+  def getUniqueKeys(relNode: RelNode, planner: StreamPlanner): Array[Array[String]] = {
+    val rowType = relNode.getRowType
+    val fmq = FlinkRelMetadataQuery.reuseOrCreate(planner.getRelBuilder.getCluster.getMetadataQuery)
+    val uniqueKeys = fmq.getUniqueKeys(relNode)
+    if (uniqueKeys != null) {
+      uniqueKeys.filter(_.nonEmpty).toList.map { uniqueKey =>
+        val keys = new util.HashSet[String]()
```

Comment:
Why a `HashSet` is needed? It seems that the indices returned by the `ImmutableBitSet` is strictly ascending. Why not just `uniqueKey.asList().toArray`?

Reply:
I'll remove `HashSet`.